src/pubsub/ua_pubsub_manager.c

```diff
@@ -778,6 +778,14 @@ UA_PubSubManager_delete(UA_Server *server, UA_PubSubManager *pubSubManager) {
         UA_Server_removeStandaloneSubscribedDataSet(server, tmpSDS1->identifier);
     }
 
+#ifdef UA_ENABLE_PUBSUB_SKS
+    /* Remove the SecurityGroups */
+    UA_SecurityGroup *tmpSG1, *tmpSG2;
+    TAILQ_FOREACH_SAFE(tmpSG1, &server->pubSubManager.securityGroups,
+                       listEntry, tmpSG2)
+        removeSecurityGroup(server, tmpSG1->securityGroupNodeId);
```

Comment:
You already have the SG. No need to forward the id and do a second lookup.
removeSecurityGroup should take a pointer to the SG.
Any user of it can do a UA_SecurityGroup_findSGbyId before.
removeSecurityGroup then also does not have to return a statuscode. It isn't evaluated anyhow.